drivers/mpsl/clock_control/clock_control_mpsl.c

```diff
@@ -207,30 +249,73 @@ static enum clock_control_status clock_get_status(struct device *dev,
 	}
 }
 
+static int clock_async_start(struct device *dev, clock_control_subsys_t subsys,
+			     struct clock_control_async_data *data)
+{
+	struct mpsl_clock_control_data *mpsl_control_data = dev->driver_data;
+	enum clock_control_status clock_status = clock_get_status(dev, subsys);
+	sys_slist_t *list = &(mpsl_control_data->async_on_list);
+
+	switch (clock_status) {
+	case CLOCK_CONTROL_STATUS_ON:
+		/* LFCLK->Clock is started by default and not able to stop it.
```

Comment:
so in case of HF request and HF clock running user gets notified and this function returns. So `clock_start` is not called. How reference counter is incremented then?

Reply:
Good catch. You are right. There is a missing. I will fix it add more tests to cover this.

Reply:
A reference counter is added in the driver to handle this case.